### PR TITLE
More cleanups

### PR DESCRIPTION
**Pull request by danwatt**
Most of these fix some sonar warnings
* Replace most printStackTrace() calls to use a logger - a few remain, they are grouped with a bunch of other System.out lines
* Remove some dead stores that were put in there a long time ago to "allow GC ASAP"
* Several classes had compareTo methods, but no equals / hashcode
* Switched a manual null check in a constructor to use `Objects.requireNonNull`
* There were a few uses of string concatenation inside of a loop, which isn't as efficient as allocating a StringBuilder (sonar  squid:S1643)

Fix the following IntelliJ warnings:
* for loop replaceable with foreach
* indexOf expression is replaceable with contains
* unnecessary boxing
* unnecessary unboxing
* Collections.sort could be replaced with List.sort
* Lambda can be replaced with method reference
* Duplicate throws

### CODE REVIEW

File: src/main/java/org/schemaspy/model/Catalog.java [hunk 1]
```diff
@@ -19,25 +19,22 @@
 package org.schemaspy.model;
 
 import java.util.Map;
+import java.util.Objects;
 
 import org.schemaspy.util.CaseInsensitiveMap;
 
 public final class Catalog implements Comparable<Catalog>{
     private String name;
```
Comment (Riggs333) [inline src/main/java/org/schemaspy/model/Catalog.java]: Consider making the field `name` final and remove the setter as it is never used (as far as I can see).

File: src/main/java/org/schemaspy/model/Catalog.java [hunk 2]
```diff
@@ -19,25 +19,22 @@
...
     private String comment = null;
-    private Map<String, Schema> schemas = new CaseInsensitiveMap<Schema>();
-    
+
 	public Catalog(String name) {
 		super();
```
Comment (Riggs333) [inline src/main/java/org/schemaspy/model/Catalog.java]: Remove the unnecessary `super()` call.

File: src/main/java/org/schemaspy/model/Catalog.java [hunk 3]
```diff
@@ -19,25 +19,22 @@
...
-		 if (name == null)
-	            throw new IllegalArgumentException("Catalog name can't be null");
+		Objects.requireNonNull(name);
 		this.name = name;
```
Comment (Riggs333) [inline src/main/java/org/schemaspy/model/Catalog.java]: It's a matter of taste but as `Objects#requireNonNull` returns the given value (unless it's `null) we can join the lines 32 and 33.

File: src/main/java/org/schemaspy/model/Catalog.java [hunk 4]
```diff
@@ -19,25 +19,22 @@
...
 	}
 	
 	public Catalog(String name, String comment) {
-		super();
-		 if (name == null)
-	            throw new IllegalArgumentException("Catalog name can't be null");
+		this(name);
 		this.name = name;
```
Comment (Riggs333) [inline src/main/java/org/schemaspy/model/Catalog.java]: As the one-argument-constructor is called here (`this(name)`) there is no need to assign the value to the field `name` again. So line 38 can be removed.

File: src/main/java/org/schemaspy/model/Catalog.java [hunk 5]
```diff
@@ -61,5 +58,16 @@ public int compareTo(Catalog i) {
...
+
+	@Override public boolean equals(Object o) {
+		if (this == o) return true;
+		if (o == null || getClass() != o.getClass()) return false;
+		Catalog catalog = (Catalog) o;
+		return Objects.equals(name, catalog.name);
+	}
+
+	@Override public int hashCode() {
+
```
Comment (Riggs333) [inline src/main/java/org/schemaspy/model/Catalog.java]: Please remove this empty line.

File: src/main/java/org/schemaspy/model/Catalog.java [hunk 6]
```diff
@@ -61,5 +58,16 @@ public int compareTo(Catalog i) {
...
+		return Objects.hash(name);
```
Comment (Riggs333) [inline src/main/java/org/schemaspy/model/Catalog.java]: IntelliJ complains on this line about "Object instantiation inside `hashCode()` (varargs call)".
Should not make that much of a difference but I propose that a `name.hashCode()` is enough here. ;-)

File: src/main/java/org/schemaspy/model/Catalog.java [hunk 7]
```diff
@@ -61,5 +58,16 @@ public int compareTo(Catalog i) {
     public String toString() {
         return name;
     }
-    
+
+	@Override public boolean equals(Object o) {
```
Comment (Riggs333) [inline src/main/java/org/schemaspy/model/Catalog.java]: Please add test(s) for this new code.
My proposal is to use the brilliant [EqualsVerifier](http://jqno.nl/equalsverifier/) that makes those tests more or less a oneliner.

File: src/test/java/org/schemaspy/model/CatalogTest.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
+package org.schemaspy.model;
+
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CatalogTest {
```
Comment (Riggs333) [inline src/test/java/org/schemaspy/model/CatalogTest.java]: My proposal is to use the brilliant [EqualsVerifier](http://jqno.nl/equalsverifier/) that makes those tests more or less a oneliner.

**Comment by danwatt** [comment]
Thanks @Riggs333 for the suggestions.

There are a lot more changes that could be done to model classes - very inconsistent use of final fields, some do a lot more than just hold values (Database), most lack tests, etc.

**Pull request merged**